drivers/serial/uart_stm32.c

```diff
@@ -1369,7 +1378,16 @@ static void uart_stm32_isr(const struct device *dev)
 		return;
 	}
 #endif /* CONFIG_UART_INTERRUPT_DRIVEN */
-	if (LL_USART_IsEnabledIT_IDLE(usart) &&
+
+	if (HAS_RTO(config->usart) && LL_USART_IsEnabledIT_RTO(usart) &&
+	    LL_USART_IsActiveFlag_RTO(usart)) {
```

Comment:
This won't compile for series not supporting RTO (L1/F1 from what I can see quickly)